google/google-how-to/gce/set-hostname-cloudinit.rst

```diff
@@ -0,0 +1,55 @@
+Set hostname with cloud-init on GCE
+===================================
+
+Hostnames can be set in many ways, including a user manually configuring the hostname, DHCP, and using a set-up tool like cloud-init.  On Ubuntu, most of these methods use the underlying ``hostname`` command, which sets the hostname statically if the ``/etc/hostname`` file exists and transiently if it does not.
+
+Historically, all Ubuntu images have an ``/etc/hostname`` file and cloud-init, when run, would create an ``/etc/hostname`` file if it did not already exist.  Starting in 24.04 GCE images, the ``/etc/hostname`` file is no longer present by default and the cloud-init key ``create_hostname_file`` set to false.  This means that when a user or tool uses ``hostname`` to set the hostname, it is set transiently and trying to set the instance hostname using solely user-supplied metadata to cloud-init or manually from the command line will be overwritten by Google's DHCP service.  It is possible to set up a GCE instance with a user defined hostname which ensures Google's DHCP-supplied hostname matches the desired hostname but if that is not done, the DHCP hostname will be a random string such as ``[INTANCE-ID].c.[PROJECT].internal``.  A common programmatic way to set hostnames for machines is to use cloud-init metadata through the gcloud flag ``--metadata-from-file user-data=[FILENAME]``, where ``[FILENAME]`` contains the cloud-init configurations desired on the VMs including the hostname.  Starting in 24.04 GCE images, only passing the ``hostname`` key to cloud-init will no longer statically set the instance hostname.
+
+
+To set the hostname using DHCP
```

Comment:
This title is a little misleading. It seems to imply that we'll be using DHCP to set the hostname (i.e. ask for and use a DNS name provided by DHCP). However, I think what we are doing is setting our hostname to a DHCP based DNS name that we already know/use.